core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1803,6 +1803,16 @@ package object config {
     .toSequence
     .createWithDefault(Nil)
 
+  private[spark] val ARCHIVES = ConfigBuilder("spark.archives")
+    .version("3.1.0")
+    .doc("Comma separated list of archives to be extracted into the working directory of each " +
+      "executor. .jar, .tar.gz, .tgz and .zip are supported. You can specify the directory " +
+      "name to unpack via adding '#' after the file name to unpack, for example, " +
+      "'file.zip#directory'. This configuration is experimental.")
```

Comment:
Ur, if this is still experimental, we should not deprecate the old ways.
> This configuration is experimental.

Reply:
Makes sense. I will instead say the behaviour will be deprecated. We can remove that back once this configuration becomes stable.